idea/tests/testData/intentions/addNameToArgument/annotationCall.kt

```diff
@@ -0,0 +1,4 @@
+annotation class A(val p1: Int, val p2: Int)
+
+@A(1, <caret>2)
```

Comment:
Maybe a dumb question, but is there any special reason that we suggest to use named argument for `p2` over `p1`?

Reply:
I didn't intend anything with it; I copied `superClassConstructor.kt` and just changed it slightly. 😛 . But I suppose this makes sure it's picking the correct mapped argument and not just the first one, or the first one that matches the type.

Reply:
It's still unclear, since both `p1` and `p2` have the same type, and both are not `vararg`.

Reply:
The test doesn't have anything to do with `vararg`... It's testing that the intention works on annotation call arguments.

I think historically the reason that in `superClassConstructor.kt`, `p2` was chosen instead of `p1` was _before_ `MixedNamedArgumentsInTheirOwnPosition`, you could only have named arguments at the end of the argument list. After that language feature, either could be used, and this is tested in `mixingPositionedAndNamed.kt`.

Anyway, I don't see a really good reason to change this particular test, even though we could go with `p1` or just go with one argument.

Reply:
I don't expect you to change the test; I'm just curious why `p2` is preferred over `p1`. I think I found an answer. Once we use `p1` with named argument, `p2` can't be positional argument anymore, i.e., `p2` should be named argument as well. And I think that's why all the following tests kinda prefer to pick the later parameter, or... un-named argument just before the named arguments start? :)

Reply:
Just to clarify, as I mentioned in my previous comment, `p1` _can_ be named by itself (e.g., `@A(p1 = 1, 2)`) when `MixedNamedArgumentsInTheirOwnPosition` is enabled (since 1.4). But I think `p2` was used since the tests were written before this feature.